src/modules/_nomodule/PixelManipulation.js

```diff
@@ -73,12 +73,13 @@ module.exports = function PixelManipulation(image, options) {
     }
 
     // perform any extra operations on the entire array:
-    if (options.extraManipulation) pixels = options.extraManipulation(pixels);
-
+    if (options.extraManipulation) res = options.extraManipulation(pixels,generateOutput);
```

Comment:
Won't this affect the other modules ?

Reply:
Nope that won't . If you go through the complete code, the function that would require generateOutput function which is just a functional version of the earlier below code this does not affect anything else.

Reply:
This might be the line which is causing problems. Have you rebased properly?

Reply:
Can we add a line or two of documentation for `extraManipulation`?